### ISSUE TITLE
Rule "vue/valid-v-for" is not working with template strings

### ISSUE THREAD

**Issue by nmaoez** [vuejs/eslint-plugin-vue#734]
**Tell us about your environment**	
* **ESLint version:**	5.0.0
* **eslint-plugin-vue version:** 5.0.0
* **Node version:** 10.13.0

**Please show your full configuration:**
```json
{
    "env": {
        "browser": true,
        "node": true,
        "es6": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:vue/recommended"
    ],
    "parserOptions": {
        "sourceType": "module",
        "parser": "babel-eslint",
        "ecmaVersion": 8
    },
    "rules": {
        "no-console": "off",
        "no-debugger": 0,
        "vue/no-unused-vars": "error",
        "vue/max-attributes-per-line": "off",
        "vue/component-name-in-template-casing": "off",
        "vue/singleline-html-element-content-newline": "off",
        "vue/html-self-closing": "off"
    }
}
```

**What did you do?**
```vue
<template>
  <div v-if="similarWords.words.length > 0" class="input-group">
    <v-chip v-for="(word, index) in similarWords.words" :key="`${index}-similar-${word}`" :id="`similar-word-${word}`">{{ word }}</v-chip>
  </div>
</template>
```

**What did you expect to happen?**
The eslint-plugin-vue complains the following directive: :key="`${index}-similar-${word}`"

But it shouldn't since both variables, which are definied in the v-for scope, are being used in the template string.
``

**What actually happened?**
Expected 'v-bind:key' directive to use the variables which are defined by the 'v-for' directive.eslint(vue/valid-v-for)

**Comment by mysticatea** [comment]
Thank you for this report.

However, I could not reproduce it. Would you make a repro case on [our online demo](https://mysticatea.github.io/vue-eslint-demo/#eJyNVMGu2yAQ/BULvUMrZZuqxyh9P9FKveTwCKxj+jC4CzhJLf97F7txnZBDL7EzM8zuDphBKK9R7MQ+YttZGfH14Kpqr01f9WDqrwcRTGuspB+edPh0nn4tulNsqtfq80FUysoQWGdclyKcyKfuICYX9ulBNaZjq9oTaz7k9ZvKOI2Xj/yoSnN23L3jlcVvL8MkHOGvCl6GLBnfssboLLkxGV/Rr8NQ5fdqHPfbuYd5ri0Pxm/77TKu2AhKFoPYDaJPuFW+bdFF0IZQRdNzOF82E/MzXCAFDNBLCgvqPMhwdQqMA17LGaCGjnyHFA3e6ZR3dQrGnWDKA3LAa16nDoFnDw+gNYo7BRkjmSP73/EdN5MtkcjTmiAMSD038+gYGkkM52a9m0aVUd4JjEbAuub5w3+MdYsyF7rHL5ErSWhTiFI1uCaT4yhXTYQn5GPODE6ZwdnEZs5woQl/Jd6x1VQmFGTuH+K1yzIXIiUVn1gQ8qkjfsRErmDnrVuPujDSGg0aa5lsnGqtJNnrWZb/bObpvAO8SBUXeHZdMibvH7kejvydFKCyXr4XKFouszp2a7wA1wHfsCa2tgCfGLZ8r5RC755AqqzcUYmFxp8LMB8yBseNyHeFi9/M73nl/Pc7bzbfbqGTXGQj8seCPJM4yiNawGCNi2L8A5H2vOU=)?

**Comment by michalsnik** [comment]
I could not reproduce it either. Although, you'll get this error if you forget template literal quotes:
```
 :key="${index}-similar-${word}"
```
if you add them however:
```
:key="`${index}-similar-${word}`"
```
it works fine

**Comment by michalsnik** [comment]
Please feel free to reopen this issue @nmaoez once you get it reproduced on our online demo.
